### DISCUSSION TITLE
Multi-handles using multiple domain names

### DISCUSSION THREAD

**Discussion by mettlex** [bluesky-social/atproto#2206] [Bluesky Lexicons]
Since a person can own multiple domain names, it should be possible to make one domain name as a primary handle and others as secondary handles.

One of the use-cases is that if one can redirect multiple domain-based handles to one account, the ownership of those multiple domain names can be proved in a social construct. This is similar to what Keybase.io had got before the project became inactive after the acquisition by Zoom.

I'm just setting up the idea here to discuss it further.

**Comment by snarfed** [comment]
Hell yes! Another use case is personal + work handles, or more generally any organizations you might have an affiliation with.

**Comment by bnewbold** [comment]
This idea has come up a few times, but there are some user experience questions to resolve and we aren't planning to move forward with this for now at the "atproto identity" layer.

The identity system aggressively re-validates the core identity metadata. This has has non-trivial resource and latency costs at scale, and moving from a single handle to multiple handles would multiply that cost, and the complexity of implementation.

We also know that we'll want to support other mechanisms for inter-linking accounts, like rel=me metadata in HTML tags (as used in Mastodon). Mixing the core identity primitives with this more application-layer interlinking is a bit tangled.

One thing we could end up doing is allow folks to configure multiple domains in their DID doc, but make it clear that there is only one "handle". The other entries would be ignored except at the application level when validating inter-linked accounts.

**Reply by mettlex** [reply]
Yeah, this makes sense. Adding multiple domains in one's DID doc should be done at scale.

**Comment by ghost** [comment]
this is good, i mentioned something similar here https://github.com/bluesky-social/social-app/issues/1085
